File: bubble-sort/c/bubble.c
#include <stdio.h>

void bubbleSort(int arr[], int n) {
    int i, j, temp;
    for (i = 0; i < n-1; i++) {     
        for (j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                // Swap arr[j] and arr[j+1]
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

int main() {
    int n, i;

    printf("Enter the number of elements: ");
    if (scanf("%d", &n) != 1) {
        printf("Error reading the number of elements.\n");
        return 1;
    }

    int arr[n];

    printf("Enter %d integers (one per line):\n", n);
    for (i = 0; i < n; i++) {
        if (scanf("%d", &arr[i]) != 1) {
            printf("Error reading an element.\n");
            return 1;
        }
    }

    bubbleSort(arr, n);

    printf("Sorted array:\n");
    for (i = 0; i < n; i++) {
        printf("%d\n", arr[i]);
    }

    return 0;
}

File: bubble-sort/ruby-dynamic-alloc/bubble.rb
def bubble_sort(arr)
  n = arr.length
  loop do
    swapped = false

    (n-1).times do |i|
      if arr[i] > arr[i+1]
        arr[i], arr[i+1] = arr[i+1], arr[i]
        swapped = true
      end
    end

    break unless swapped
  end
  arr
end

puts "Enter the number of elements:"
n = gets.to_i

arr = []
puts "Enter #{n} integers (one per line):"
n.times do
  arr << gets.to_i
end

sorted_array = bubble_sort(arr)

puts "Sorted array:"
puts sorted_array

